app/src/main/java/com/codesoom/assignment/controllers/TaskController.java

```diff
@@ -0,0 +1,60 @@
+package com.codesoom.assignment.controllers;
+
+import com.codesoom.assignment.models.Task;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+
+public class TaskController {
+    private final HashMap<Long, Task> tasks = new HashMap<>();
+    private Long curTaskID = 0L;
+
+    public List<Task> getAllTask(){
+        List<Task> AllTasks = new ArrayList<>();
+
+        for(Task item : tasks.values()){
+            AllTasks.add(item);
+        }
+        return AllTasks;
```

Comment:
```suggestion
        return tasks.values();
```
위 코드와는 어떻게 다른지 설명해주세요!

Reply:
위의 코드는 List를 반환하고, 아래 코드는 Collections를 반환합니다!

List는 Collections의 하위 집합이므로 new ArrayList<>(tasks.values())를 담아 return하는  방법으로 고쳤습니다.

Reply:
좋습니다! 
혹시 메소드 리턴값을 `Collection<Task>` 로 하면 어떻게 될까요?

Reply:
Collection<Task>를 리턴값으로 해도 List<Task>와 큰 차이점은 나지 않았습니다!

Reply:
- 오 task가 입력받은 순서가 보장되어야 한다는 내용이 필수라고 생각된다면  그런 중요한 내용은 주석에 추가해놓는 것이 정말 중요합니다!
- 지금은 기존에 방식인 List로 리턴을 하더라도 순서를 보장하는지 확신할 수 없습니다.
  처음부터 `task.values()` 는 List가 아닌 Collections 을 리턴하고 있기 때문입니다. 그것을 List로 넣는다고 입력순서가 보장될 순 없습니다.

Reply:
- Collection 인터페이스를 직접 관찰하고 주요 역할이 무엇인지 알려주세요!
https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html

Reply:
> This interface is typically used to pass collections around and manipulate them where maximum generality is desired.

Reply:
좋습니다. 그럼 다시 넘어가서 순서를 보장하기 위해 우리는 다른 일을 해야 합니다.

어떤 순서가 보장되어야 하나요? 그것을 주석으로 남겨보세요!
그리고 가능하면 구현도 해보시는 것이 좋습니다!

Reply:
`b16a0bf` 해당 함수에서 ID순으로 List를 정렬하는 Compatator를 사용하여 입력 순서를 보장하는 List를 구현했습니다.